homeassistant/components/sensor/sochain.py

```diff
@@ -0,0 +1,94 @@
+"""
+Support for watching multiple cryptocurrencies.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.sochain/
+"""
+import asyncio
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_NAME, ATTR_ATTRIBUTION)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+REQUIREMENTS = ['python-sochain-api==0.0.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ADDRESS = 'address'
+CONF_NETWORK = 'network'
+CONF_ATTRIBUTION = "Data provided by chain.so"
+
+DEFAULT_NAME = 'Crypto Balance'
+
+SCAN_INTERVAL = timedelta(minutes=5)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ADDRESS): cv.string,
+    vol.Required(CONF_NETWORK): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the sochain sensors."""
+    address = config.get(CONF_ADDRESS)
+    network = config.get(CONF_NETWORK)
+    name = config.get(CONF_NAME)
+
+    connection = SochainSensor(hass, name, network, address)
+    yield from connection.async_update()
+
+    if connection.state is None:
+        _LOGGER.error("Check that your network and address is correct")
+        return False
+
+    async_add_devices([connection])
+
+
+class SochainSensor(Entity):
+    """Representation of a Sochain sensor."""
+
+    def __init__(self, hass, name, network, address):
+        """Initialize the sensor."""
+        from pysochain import ChainSo
+        self.hass = hass
+        self._name = name
+        self._unit_of_measurement = network.upper()
+        self._websession = async_get_clientsession(self.hass)
+        self.chainso = ChainSo(
```

Comment:
The `swiss_public_transport` sensor is not the best way to do it. Sorry, my bad.

Please move this to `async_setup_platform()` and just pass the data object down.

```python
    session = async_get_clientsession(hass)
    api = ChainSO(network, address hass.loop, session)
...
    async_add_devices([SochainSensor(name, api)], True)
```